certora/specs/ExchangeRate.spec

```
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (c) 2025 Morpho Association
+
+import "Invariants.spec";
+
+definition shares() returns mathint = currentContract.totalSupply + currentContract.virtualShares;
+
+definition assets() returns mathint = currentContract._totalAssets + 1;
+
+// Check that if deposit adds one more share to the user than it does, then the share price would decrease following a deposit.
```

i find it more meaningful
```suggestion
// Show that the share price increase is bounded by a/(s(s+1)).
```

---

I don't understand what this is saying, what are `a` and `s`, and why do we look at `a/(s(s+1))`.

To me the property we want to show here is that the protocol does the best it can for the user. So for a given amount of assets passed as inputs, the shares are increased as much as possible such that the share price still increases. It shows that the rounding is perfect in that sense (can't do better). Maybe there is a better way to phrase it ?

---

a=assets, s=shares. I quickly made the maths and I think that this is equivalent to what we are proving, but the property is clearer: the share price increase is not more than something small